src/main/java/no/nav/pto/veilarbportefolje/vedtakstotte/VedtakService.java

```diff
@@ -1,32 +1,75 @@
 package no.nav.pto.veilarbportefolje.vedtakstotte;
 
+import io.vavr.Tuple2;
+import lombok.extern.slf4j.Slf4j;
 import no.nav.pto.veilarbportefolje.domene.AktoerId;
+import no.nav.pto.veilarbportefolje.domene.Fnr;
 import no.nav.pto.veilarbportefolje.elastic.ElasticIndexer;
+import no.nav.pto.veilarbportefolje.service.AktoerService;
+
+@Slf4j
 public class VedtakService {
 
     private VedtakStatusRepository vedtakStatusRepository;
     private ElasticIndexer elasticIndexer;
+    private AktoerService aktoerService;
 
-    public VedtakService(VedtakStatusRepository vedtakStatusRepository, ElasticIndexer elasticIndexer) {
+    public VedtakService(VedtakStatusRepository vedtakStatusRepository, ElasticIndexer elasticIndexer, AktoerService aktoerService) {
         this.vedtakStatusRepository = vedtakStatusRepository;
         this.elasticIndexer = elasticIndexer;
+        this.aktoerService = aktoerService;
     }
 
     public void behandleMelding(KafkaVedtakStatusEndring melding) {
         KafkaVedtakStatusEndring.KafkaVedtakStatus vedtakStatus = melding.getVedtakStatus();
         switch (vedtakStatus) {
-            case UTKAST_SLETTET:
-                vedtakStatusRepository.slettVedtakUtkast(melding.getVedtakId());
-                break;
+            case UTKAST_SLETTET : {
+                slettUtkast(melding);
+                return;
+            }
             case UTKAST_OPPRETTET:
-            case SENDT_TIL_BESLUTTER:
-                vedtakStatusRepository.upsertVedtak(melding);
-                break;
-            case SENDT_TIL_BRUKER:
-                vedtakStatusRepository.slettGamleVedtakOgUtkast(melding.getAktorId());
-                vedtakStatusRepository.upsertVedtak(melding);
-                break;
+            case SENDT_TIL_BESLUTTER: {
+                oppdaterUtkast(melding);
+                return;
+            }
+            case SENDT_TIL_BRUKER: {
+                setUtkastTilSendt(melding);
+            }
         }
-        elasticIndexer.indekser(AktoerId.of(melding.getAktorId()));
     }
+
+    private void slettUtkast(KafkaVedtakStatusEndring melding) {
+        vedtakStatusRepository.slettVedtakUtkast(melding.getVedtakId());
+        skrivNullFeldterTilIndeks(melding);
+    }
+
+
+    private void oppdaterUtkast(KafkaVedtakStatusEndring melding) {
+        vedtakStatusRepository.upsertVedtak(melding);
+        skrivOppdateringTilIndeks(melding);
+    }
+
+
+    private void setUtkastTilSendt(KafkaVedtakStatusEndring melding) {
+        vedtakStatusRepository.slettGamleVedtakOgUtkast(melding.getAktorId());
+        vedtakStatusRepository.upsertVedtak(melding);
+        skrivNullFeldterTilIndeks(melding);
+    }
+
+    private void skrivOppdateringTilIndeks(KafkaVedtakStatusEndring melding) {
```

Comment:
kan detta gå i benen på deltaindekseringen

Reply:
Tror ES er first-come-first-serve så om et dokument får to samtidige writes tror jeg de blir lagt i kø og så bare overskriver den andre den første. Men er ikke 100% sikker her